tests/unit/test_kdf.cpp

```diff
@@ -0,0 +1,197 @@
+/*
+ * #%L
+ * %%
+ * Copyright (C) 2020 BMW Car IT GmbH
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+#include "util.cpp"
+
+#include "mococrw/kdf.h"
+#include "mococrw/hash.h"
+
+using namespace mococrw;
+
+using testing::Eq;
+
+struct inputData
+{
+    openssl::DigestTypes hashFunction;
+    std::vector<uint8_t> password;
+    std::vector<uint8_t> salt;
+    uint iterations;
+    uint olen;
+    std::string expectedRestult;
+};
+
+void testPbkfd2(struct inputData data)
+{
+    auto pbkfd2 = PBKDF2(data.hashFunction, data.iterations);
+    ASSERT_THAT(utility::toHex(pbkfd2.deriveKey(data.password, data.olen, data.salt)),
+                Eq(data.expectedRestult));
+}
+
+void testX963kdf(struct inputData data)
+{
+    auto x963kdf = X963KDF(data.hashFunction);
+    ASSERT_THAT(utility::toHex(x963kdf.deriveKey(data.password, data.olen, data.salt)),
+                Eq(data.expectedRestult));
+}
+
+TEST(pbkdf2_sha1, tests)
```

Comment:
please use TEST_P to avoid these lengthy tests that test 3 times the same thing

Reply:
That was easy :)